src/routes/api/profile.js

```diff
@@ -0,0 +1,30 @@
+import { Router } from 'express';
+import ProfileController from '../../controllers/profileController';
+import middleware from '../../middlewares';
+
+const { Authenticate, validate } = middleware;
+
+const router = Router();
+
+router.route('/')
+  .post(
+    Authenticate.verifyToken, ProfileController.upsertProfile
+  )
+  .get(
+    Authenticate.verifyToken, Authenticate.verifySuperOrManager, ProfileController.getAllProfiles
+  );
+  
+router.route('/:userId')
+  .get(
+    Authenticate.verifyToken, validate('checkUserId'), Authenticate.verifyProfileRights, ProfileController.getProfile
+  )
+  .put(
```

Comment:
I don't think a user should pass an id when they want to edit their profile. It should be similar to creating a profile. When updating profile whatever user that is signed in should be updated

Reply:
> I don't think a user should pass an id when they want to edit their profile. It should be similar to creating a profile. When updating profile whatever user that is signed in should be updated

Passing the `id` is adhering to REST principles of specifying the `id` of the resource to manipulate. The way the code is, it will work if the person doesn't specify the `id`. It will work, but I don't think it should be that way.

Reply:
Moreover you know this is actually an API that will be consumed by a frontend app. It's important to adhere to principles and standards, and also, the user won't be the one specifying their `id`. It will be implemented in code.